README.md

```diff
@@ -16,54 +16,104 @@ compiled correctly.
 The library is known to compile on various Linux versions (Redhat,
 Mageia, Ubuntu, Debian), Mac OS X, and Windows (Mingw-w64).
 
-Step 1: Installing LLVM
-=======================
 
```

Comment:
I've tried this pull request and it seems to compile and run fine, but I'm wondering if there is a problem with the output.

There seems to be forward references in the Global Variable Definitions and Initialization section:

Notice "_OC_str" is being referred to before it's defined below.

`/* Global Variable Definitions and Initialization */
struct l_array_3_uint8_t_KC_ mem_name = { { ((&_OC_str.array[((int32_t)0)])), ((&_OC_str_OC_1.array[((int32_t)0)])), ((&_OC_str_
OC_2.array[((int32_t)0)])) } };
static struct l_array_7_uint8_t _OC_str = { "Static" };
static struct l_array_5_uint8_t _OC_str_OC_1 = { "Heap" };
static struct l_array_6_uint8_t _OC_str_OC_2 = { "Stack" };
`